src/task.ts

```diff
@@ -244,6 +245,7 @@ export abstract class CompoundTask extends DFTask {
         // If the task has no children, then it's completed by definition.
         if (children.length == 0) {
             this.state = TaskState.Completed;
+            // TODO: how should isPlayed change here?
```

Comment:
There is still a TODO here?

Reply:
Yeah, but I think that `TODO` is not directly related to the bug we're fixing here :)
Essentially, that TODO is about how to update the Task's isPlayed flag when it is either a WhenAll or WhenAny with no sub-tasks. I haven't fully thought about whether the resulting Task should have `isPlayed` equals to `true` or `false`, especially as that will affect the `context.IsReplaying` flag.

So the way I'm thinking about this is simply as a note for us to investigate later, but it's not a remaining "to do" item for this bug fix. Does that seem reasonable?

Reply:
Your reasoning does sound reasonable yes, but I wouldn't want to merge code with a TODO in it. It's very easy to lose track of it and lose context. (there's already a couple places in the code where I've run into a TODO and not sure what it was intending to do). That being said, in order to not block this PR, I'd be more in favor of creating an issue for us to investigate this further (basically just saying what you wrote above), and removing this comment from the PR

Reply:
Also, I'm curious, what is the current behavior of this scenario (the `isPlayed` flag of a compound task that has no children)? Does this PR change the current behavior of this scenario?

Reply:
I've created that issue [here](https://github.com/Azure/azure-functions-durable-js/issues/423) and will remove the `TODO` comment on my next commit.

As for your question, the `isPlayed` flag today remains with its default value of `False`. This is an arbitrary and accidental choice, which is why I added the `TODO` comment to investigate further.

Reply:
Sounds good, thank you!